CONTRIBUTING.md

```diff
@@ -0,0 +1,161 @@
+# Contributing / Developer Guide
+
+This repository follows a set of standard development practices for modern Python packages.
+
+## Development workflow
+
+### Setting up
+1. Clone the repository:
+    ```
+    git clone https://github.com/talmolab/sleap-io && cd sleap-io
+    ```
+2. Install the package in development mode:
+   With [`conda`](https://docs.conda.io/en/latest/miniconda.html) (*recommended*):
+   ```
+   conda env create -f environment.yml && conda activate sleap-io
+   ```
+   Or without conda:
+   ```
+   pip install -e .[dev]
+   ```
+3. Test that things are working:
+   ```
+   pytest tests
+   ```
+
+To reinstall the environment from scratch using `conda`:
+```
+conda env remove -n sleap-io && conda env create -f environment.yml
+```
+
+We also recommend setting up `black` to run automatically in your IDE.
+A good way to do this without messing up your global environment is to use a tool like [`pipx`](https://pypa.github.io/pipx/):
+```
+pip install pipx
+pipx ensurepath
+pipx install black
+```
+This will make `black` available everywhere (such as VSCode), but will not be dependent on your conda `base` environment.
```

Comment:
Do we also want to setup `mypy` in the IDE?

Reply:
Yeah we could probably add a whole section describing a sample VSCode configuration but that probably warrants its own PR with screenshots/config JSONs/etc.